Gemfile

```diff
@@ -65,3 +65,18 @@ instance_eval(ENV["GEMFILE_MOD"]) if ENV["GEMFILE_MOD"]
 # If you want to load debugging tools into the bundle exec sandbox,
 # add these additional dependencies into Gemfile.local
 eval_gemfile(__FILE__ + ".local") if File.exist?(__FILE__ + ".local")
+
+# These lines added for Windows development only.
+# For FFI to call into PowerShell we need the binaries and assemblies located
+# in the Ruby bindir.
+#
+# We copy (and overwrite) these files every time "bundle <exec|install>" is
+# executed, just in case they have changed.
+if RUBY_PLATFORM =~ /mswin|mingw|windows/
+  instance_eval do
+    ruby_exe_dir = RbConfig::CONFIG["bindir"]
+    assemblies = Dir.glob(File.expand_path("distro/ruby_bin_folder", Dir.pwd) + "/*.dll")
+    FileUtils.cp_r assemblies, ruby_exe_dir, verbose: true unless ENV['_BUNDLER_WINDOWS_DLLS_COPIED']
+    ENV['_BUNDLER_WINDOWS_DLLS_COPIED'] = "1"
+  end
+end
```

Comment:
this seems a little odd if i'm reading it correctly...

but the problem is that we need these dlls in the PATH or in pwd in order to have them link correctly?  and since we might be in some other directory we don't want to copy to pwd, so we copy to `RbConfig::CONFIG["bindir"]` into the ruby?

it might be better to use __FILE__ to construct a full path to the DLLs and use that path in `FFI.ffi_lib` and maybe `Fiddle.dlopen` the dependent DLLs first to load them into the process before the ffi_lib call?

Reply:
Yes, we could load the wrapper from any location inside the distribution, but the .NET assemblies must be loaded from the location where ruby.exe is running (or we need to deploy the assemblies into the .NET Global Assembly Cache)

Reply:
right, windoze is gonna windoze.... carry on then..